### DISCUSSION TITLE
Why the benchmark data shows non-secure ops/s is much smaller than the default secure mode?

### DISCUSSION THREAD

**Discussion by liber-gong** [ai/nanoid#416] [Q&A]
nanoid                     5,652,148 ops/sec
vs
nanoid/non-secure          2,744,615 ops/sec

How should I understand the "faster non-secure generator"?

**Comment by ai** [comment]
It was faster. Maybe we need to update docs.

**Reply by ai** [reply]
I fixed the docs c79ee96

**Reply by michaelnwani** [reply]
Hi!

FYI the benchmark section still shows nanoid as faster than nanoid/non-secure:
- nanoid => 5,652,148 ops/sec
- nanoid/non-secure => 2,744,615 ops/sec

<img width="841" alt="image" src="https://github.com/ai/nanoid/assets/1715524/3602c8d4-5e85-42b7-916a-00047e696661">

**Reply by ai** [reply]
I updated not a benchmark but docs (removed that `non-secure is faster`).

Yes, secure version is faster because we added many memory tricks there.
